elf/io/intern_wrapper.py

```diff
@@ -0,0 +1,89 @@
+from collections.abc import Mapping
+
+try:
+    from intern import array
+except ImportError:
+    intern = None
+
+
+def _check_intern_importable():
+    if intern is None:
+        raise ImportError(
+            "Could not import the `intern` library. This means you cannot "
+            "download or upload cloud datasets. To fix this, you can install "
+            "intern with: \n\n\t"
+            "pip install intern"
+        )
+    return True
+
+
+class InternDataset:
+    def __init__(self, cloud_path):
+        _check_intern_importable()
+        self._data = array(cloud_path)
+
+    @property
+    def dtype(self):
+        return self._data.dtype
+
+    @property
+    def ndim(self):
+        return 3  # todo: this COULD be 4 etc...
+
+    # TODO chunks are arbitrary, how do we handle this?
+    @property
+    def chunks(self):
```

Comment:
Is there some chunking that should be observed? (Even if it's not exposed as `chunks` by the intern API?)

Reply:
We ran some benchmarks and found that there's not an enormous difference between "cuboid-aligned" and "non-aligned" reads with BossDB in terms of performance, because of the server-side cache. We can add the 512²64 chunks here but it's luckily not a big contributor/detractor to performance! I figured it made more sense to remain "agnostic" here in the same way MRC does.

Reply:
And are parallel writes to data with overlapping chunks ok? If yes, and if performance is not an issue, we can return `None` indeed.
(Note that this will need some updates in `cluster_tools` then, but I think it's better to update it there so that it can deal with arbitrary chunk sizes rather than adding an artificial one here.)